lib/erb_lint/cli.rb

```diff
@@ -99,13 +117,49 @@ def run(args = ARGV)
 
     private
 
+    attr_reader :cache, :config
+
+    def run_on_file(runner, filename)
+      file_content = read_content(filename)
+
+      if with_cache? && !autocorrect?
+        run_using_cache(runner, filename, file_content)
+      else
+        file_content = run_with_corrections(runner, filename, file_content)
+      end
+
+      log_offense_stats(runner, filename)
+      file_content
+    end
+
+    def run_using_cache(runner, filename, file_content)
+      if cache.include?(filename)
+        runner.restore_offenses(cache.get(filename, file_content))
```

Comment:
Could we avoid the `File.exist?` then `File.read` by removing `Cache#include?` and relying on `Cache#get` returning `nil` if there's no file? That skips one system call, we just open the file, instead of doing a `fstat` then `open`.

Reply:
Yeah, I tried to avoid it, and this lead me to explore a few other optimizations and places that we can avoid repeatedly `read`-ing or `stat`-ing the file. For instance, when calculating the checksum, we again read the file if you look at the source of `file` on `Digest::SHA1` class:
https://ruby-doc.org/stdlib-2.4.0/libdoc/digest/rdoc/Digest/Instance.html#method-i-file

this also unfortunately led to the demise of the `#[]=` syntax for the getter but I think it's worth it for clarity, unless you can think of other ways to do this.

Check out the latest commit for the changes!